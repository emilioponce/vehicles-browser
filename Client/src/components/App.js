import React, { Component } from 'react'
import Header from './Header'
import VehicleList from './VehicleList'
import VehicleMap from './VehicleMap'
import AppProvider from '../config/AppProvider'
import { UBER_TYPE } from '../config/constants'
import { getTaxis, getUbers } from '../api'

import styles from './App.module.css'

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      vehicles: []
    }
  }

  componentDidMount() {
    Promise.all([getTaxis(), getUbers()])
      .then(([taxis, ubers]) => {
        // taxis have type, ubers don't. We add the type.
        let ubersWithType = ubers.data.placemarks.map(car => {
          car.type = UBER_TYPE
          return car
        })

        // unique array for both kind of vehicles
        let vehicles = [...taxis.data.poiList, ...ubersWithType]
        this.setState({
          vehicles
        })
      })
      .catch(error => console.log('Error retrieving cars'))
  }

  render() {
    const { vehicles } = this.state
    return (
      <div id={styles.page}>
        <div className={styles.header}>
          <Header />
        </div>
        <AppProvider>
          <div className={styles.list}>
            <VehicleList vehicles={vehicles} />
          </div>
          <div className={styles.map}>
            <VehicleMap />
          </div>
        </AppProvider>
      </div>
    )
  }
}

export default App
