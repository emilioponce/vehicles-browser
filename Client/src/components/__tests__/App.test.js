import React from 'react'
import ReactDOM from 'react-dom'
import App from '../App'
import renderer from 'react-test-renderer'

jest.mock('../VehicleMap')

const component = <App />

describe('Component App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(component, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('snapshot renders correctly', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
