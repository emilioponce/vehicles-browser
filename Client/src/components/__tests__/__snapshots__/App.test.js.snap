// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component App snapshot renders correctly 1`] = `
<div
  id="page"
>
  <div
    className="header"
  >
    <div
      className="wrapper"
    >
      <div
        className="header"
      >
        <img
          alt="Vehicles Browser"
          className="logo"
          src="logo.svg"
        />
        <div
          className="title"
        >
          Vehicles Browser
        </div>
        <div
          className="subtitle"
        >
          A tool for visualizing different types of vehicles
        </div>
      </div>
    </div>
  </div>
  <div
    className="list"
  >
    <div
      className="vehicleList"
    >
      Loading...
    </div>
  </div>
  <div
    className="map"
  />
</div>
`;
